specs/Extensions.md

```diff
@@ -0,0 +1,604 @@
+# Background
+Developers would like to ensure a browser extension is installed so that they may take advantage of the functionality the extension is providing.
+
+# Description
+Enable extension services in WebView2. Then end developer can `AddBrowserExtension` from folder path, `GetBrowserExtension` to get list of extensions installed. Once an extension is installed,
+developers can use `ICoreWebView2BrowserExtension` to get id, name of this extension. And also remove or enable/disable this extension.
+
+# Examples
+
+The following code snippet demonstrates how the Extensions related API can be used:
+
+## Win32 C++
+
+```cpp
+static constexpr WCHAR c_samplePath[] = L"extensions/example-devtools-extension";
+
+void AppWindow::InitializeWebView()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    CHECK_FAILURE(options->put_AreBrowserExtensionsEnabled(TRUE));
+
+    // ... other option properties
+
+    // ... CreateCoreWebView2EnvironmentWithOptions
+
+    InstallDefaultExtensions();
```

Comment:
It would be good to separate the two scenarios (End dev installs app extension on first run, End user manages extensions) into their own sections with their own C++ and C# code.